src/components/access-type.js

```diff
@@ -320,16 +322,36 @@ class AccessTypeBase extends React.Component {
         };
   }
   //TODO -> need to write test cases to cover all scenarios , selectedPlan, planType , coupon, urls, story details etc.
-  initRazorPayPayment = (selectedPlanObj = {}, planType = "", storyId = "", storyHeadline = "", storySlug = "") => {
+  initRazorPayPayment = async (
+    selectedPlanObj = {},
+    planType = "",
+    storyId = "",
+    storyHeadline = "",
+    storySlug = "",
+    paymentType = ""
+  ) => {
     if (!selectedPlanObj) {
       console.warn("Razor pay needs a plan");
       return false;
     }
 
     const planObject = this.makePlanObject(selectedPlanObj, planType, storyId, storyHeadline, storySlug); //we are doing this to sake of backward compatibility and will be refactored later.
-    const { paymentOptions } = this.props;
-    planObject["paymentType"] = get(planObject.selectedPlan, ["recurring"]) ? "razorpay_recurring" : "razorpay";
-    const paymentObject = this.makePaymentObject(planObject);
+    let { paymentOptions } = this.props;
+    const { discounted_price_cents, price_cents } = planObject.selectedPlan;
+
+    if (discounted_price_cents === 0 || price_cents === 0) {
+      const { data } = await awaitHelper(global.AccessType.getPaymentOptions(0));
+      paymentOptions = data;
+    }
+
+    planObject["paymentType"] =
+      paymentType || (get(planObject.selectedPlan, ["recurring"]) ? "razorpay_recurring" : "razorpay");
```

Comment:
Can we change `get(planObject.selectedPlan, ["recurring"]) ` to `get(planObject, ["selectedPlan", "recurring"])`